### PR TITLE
feat(EMS-2367): No PDF - Your buyer - Prior trading history redirect based on total contract value

### PR DESCRIPTION
**Pull request by Zainzzkk**
## Introduction :pencil2:
This PR adds redirect based on if total contract value is below or above threshold 

## Resolution :heavy_check_mark:
* Added middleware for checking if totalContractValue.value is above or below threshold and adding local application flag
* Added to prior-trading-history controller for redirect
* Added tests

## Miscellaneous :heavy_plus_sign:
* Added flag for createAnApplication e2e test command for totalContractValue

### CODE REVIEW

File: e2e-tests/commands/insurance/complete-buyer-section.js [hunk 1]
```diff
@@ -4,14 +4,28 @@
...
   cy.completeAndSubmitCompanyOrOrganisationForm({});
   cy.completeAndSubmitConnectionToTheBuyerForm({});
   cy.completeAndSubmitTradedWithBuyerForm({ exporterHasTradedWithBuyer });
 
+  /**
+   * if not exporterHasTradedWithBuyer
+   * and totalContractValueOverThreshold is true
+   * then complete credit insurance cover form
+   */
+  if (totalContractValueOverThreshold && !exporterHasTradedWithBuyer) {
```
Comment (ttbarnes) [inline e2e-tests/commands/insurance/complete-buyer-section.js:23]: Documentation is the other way around to the condition

File: e2e-tests/insurance/cypress/e2e/journeys/your-buyer/check-your-answers/check-your-answers-page.spec.js [hunk 2]
```diff
@@ -52,7 +53,7 @@ context('Insurance - Your buyer - Check your answers - As an exporter, I want to
     cy.corePageChecks({
       pageTitle: CONTENT_STRINGS.PAGE_TITLE,
       currentHref: `${ROOT}/${referenceNumber}${CHECK_YOUR_ANSWERS}`,
-      backLink: `${ROOT}/${referenceNumber}${TRADED_WITH_BUYER}`,
+      backLink: `${ROOT}/${referenceNumber}${BUYER_FINANCIAL_INFORMATION}`,
```
Comment (ttbarnes) [inline e2e-tests/insurance/cypress/e2e/journeys/your-buyer/check-your-answers/check-your-answers-page.spec.js:56]: Is the back link now conditional depending on the total contract value threshold?

File: e2e-tests/insurance/cypress/e2e/journeys/your-buyer/traded-with-buyer/traded-with-buyer-total-contract-value-over-threshold.spec.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      referenceNumber = refNumber;
+
+      cy.startInsuranceYourBuyerSection({});
+
+      cy.completeAndSubmitCompanyOrOrganisationForm({});
+      cy.completeAndSubmitConnectionToTheBuyerForm({});
+
+      url = `${Cypress.config('baseUrl')}${INSURANCE_ROOT}/${referenceNumber}${TRADED_WITH_BUYER}`;
+      tradingHistoryUrl = `${Cypress.config('baseUrl')}${INSURANCE_ROOT}/${referenceNumber}${TRADING_HISTORY}`;
+      creditInsuranceCoverUrl = `${Cypress.config('baseUrl')}${INSURANCE_ROOT}/${referenceNumber}${CREDIT_INSURANCE_COVER}`;
```
Comment (ttbarnes) [inline e2e-tests/insurance/cypress/e2e/journeys/your-buyer/traded-with-buyer/traded-with-buyer-total-contract-value-over-threshold.spec.js:27]: `baseUrl` const just before `context`? DRY

File: src/ui/server/controllers/insurance/your-buyer/traded-with-buyer/index.test.ts [hunk 4]
```diff
@@ -195,6 +203,23 @@ describe('controllers/insurance/your-buyer/traded-with-buyer', () => {
...
         });
       });
+
+      describe(`when the totalContractValueOverThreshold is true and ${TRADED_WITH_BUYER} is false`, () => {
+        it(`should redirect to ${CREDIT_INSURANCE_COVER}`, async () => {
+          req.body = {
+            [TRADED_WITH_BUYER]: 'false',
+          };
+
+          // @ts-ignore
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/your-buyer/traded-with-buyer/index.test.ts:213]: Why `ts-ignore`? Can update application types?

Comment (Zainzzkk) [reply src/ui/server/controllers/insurance/your-buyer/traded-with-buyer/index.test.ts:213]: it keeps insisting that res.local.applications could be undefined

File: e2e-tests/insurance/cypress/e2e/journeys/your-buyer/check-your-answers/check-your-answers-page.spec.js (hunk 2 above)
Comment (Zainzzkk) [reply e2e-tests/insurance/cypress/e2e/journeys/your-buyer/check-your-answers/check-your-answers-page.spec.js:56]: no, that page will always be completed

File: src/ui/server/index.ts [hunk 5]
```diff
@@ -122,6 +123,7 @@ ui.use('/insurance/eligibility', requiredInsuranceEligibilityDataProvided);
 ui.use('/insurance/:referenceNumber/*', getApplication);
 ui.use('/insurance/:referenceNumber/*', applicationAccess);
 ui.use('/insurance/:referenceNumber/*', applicationStatus);
+ui.use('/insurance/:referenceNumber/*', totalContractValueOverThreshold);
```
Comment (ttbarnes) [inline src/ui/server/index.ts:126]: Could we have this just after `getApplication`?

This new middleware could even be part of `getApplication` with a new mapping function that adds `totalContractValueOverThreshold` - not a blocker, but if we add another flag I think we should do this

File: src/ui/server/controllers/insurance/your-buyer/traded-with-buyer/index.test.ts (hunk 4 above)
Comment (ttbarnes) [reply src/ui/server/controllers/insurance/your-buyer/traded-with-buyer/index.test.ts:213]: Try this?

```js
res.locals.application = {
  ...mockApplication,
  totalContractValueOverThreshold: true
}
```

**Review by ttbarnes — APPROVED** [review]

**Pull request merged**